src/ql/pass/map/qubits/place_mip/place_mip.cc

```diff
@@ -90,81 +93,77 @@ PlaceQubitsPass::PlaceQubitsPass(
     const utils::Ptr<const pmgr::Factory> &pass_factory,
     const utils::Str &instance_name,
     const utils::Str &type_name
-) : pmgr::pass_types::KernelTransformation(pass_factory, instance_name, type_name) {
+) : pmgr::pass_types::Transformation(pass_factory, instance_name, type_name) {
     options.add_int(
         "horizon",
         "When specified, the placement algorithm will only consider the "
-        "connectivity required to perform the first N two-qubit gates of each "
-        "kernel. When not specified or zero, all two-qubit gates are "
+        "connectivity required to perform the first N most frequent two-qubit gate types. "
+        "When not specified or zero, all two-qubit gate types are "
         "considered.",
-        "0", 0, 100
+        "0", 0, utils::MAX
+    );
+    options.add_real(
+        "timeout",
+        "A float duration in seconds after which the MIP problem resolution by HiGHS will terminate. "
+        "If this happens, the pass will emit a warning and will not update the qubit indices. "
+        "When set to 0, there is no time limit.",
```

Comment:
So it will not use a mapping that exists at the time out. That is fine. This answers comment above. But there could be an option to stop the compiler. This is useful in benchmarking when silent continuation after a time out would not be desired. It is silent although a warning is given since this warnings drowns in the set of warnings during benchmarking.

Reply:
Agreed, and I think it is in fact better to BY DEFAULT stop the compiler on timeout. So the new option fail_on_timeout is TRUE by default.